dkg-gadget/src/gossip_engine/network.rs

```diff
@@ -328,43 +349,53 @@ impl<B: Block + 'static> GossipHandler<B> {
 	pub async fn run(mut self) {
 		let stream = self.my_channel.subscribe();
 		let mut incoming_messages = tokio_stream::wrappers::BroadcastStream::new(stream);
+		let mut event_stream =
+			self.event_stream.take().expect("Event stream is only taken in `run` once; qed");
 		debug!(target: "dkg_gadget::gossip_engine::network", "Starting the DKG Gossip Handler");
-		// a ticker timer to periodically check for new messages in the queue.
-		let mut timer = tokio::time::interval(core::time::Duration::from_millis(50));
-		loop {
-			futures::select! {
-				network_event = self.event_stream.next().fuse() => {
-					if let Some(network_event) = network_event {
-						self.handle_network_event(network_event).await;
-					} else {
-						// Networking has seemingly closed. Closing as well.
-						return;
-					}
-				},
-				message = incoming_messages.next().fuse() => {
-					match message {
-						Some(Ok(ToHandler::SendMessage { recipient, message })) => self.send_signed_dkg_message(recipient, message),
-						Some(Ok(ToHandler::Gossip(v))) => self.gossip_message(v),
-						None => {
-							// The broadcast stream has been closed.
-							return;
-						},
-						_ => {},
-					}
-				},
-				_tick = timer.tick().fuse() => {
-					// check if we have any messages in our queue.
-					let read_lock = self.message_queue.read();
-					if !read_lock.is_empty() {
-						// notify the caller that we have messages in the queue.
-						self.message_notifications_channel.send(()).unwrap();
-					}
-				},
+
+		let self0 = self.clone();
+		let incoming_messages_task = tokio::spawn(async move {
+			while let Some(message) = incoming_messages.next().await {
+				match message {
+					Ok(ToHandler::SendMessage { recipient, message }) =>
+						self0.send_signed_dkg_message(recipient, message),
+					Ok(ToHandler::Gossip(v)) => self0.gossip_message(v),
+					_ => {},
+				}
 			}
-		}
+		});
+
+		// a timer that fires every few ms to check if there are messages in the queue, and if so,
+		// notify the listener.
+		let self1 = self.clone();
+		let mut timer = tokio::time::interval(core::time::Duration::from_millis(100));
```

Comment:
What are the downsides of firing so often? Are there any?

Reply:
Not at all, this task running on its own, there is no blocking nor it should affect any other tasks that are running along side it, even more this task could be running on totally separate thread, the runtime decides that for us.